coderd/database/dbauthz/dbauthz.go

```diff
@@ -1754,6 +1754,19 @@ func (q *querier) GetUserLinksByUserID(ctx context.Context, userID uuid.UUID) ([
 	return q.db.GetUserLinksByUserID(ctx, userID)
 }
 
+func (q *querier) GetUserWorkspaceBuildParameters(ctx context.Context, params database.GetUserWorkspaceBuildParametersParams) ([]database.GetUserWorkspaceBuildParametersRow, error) {
+	u, err := q.db.GetUserByID(ctx, params.OwnerID)
+	if err != nil {
+		return nil, err
+	}
+	// The ability to update the user implies either the user themselves or someone
+	// with complete admin access to the user account.
+	if err := q.authorizeContext(ctx, rbac.ActionUpdate, u.UserDataRBACObject()); err != nil {
```

Comment:
I don't think we should be overloading RBAC verb/resource combos like this, as it makes what permission different roles have hard to understand.

The "natural" permission is Read on the Workspace Build, but that has a more expensive implementation since we would need to query and filter builds, then get their parameters.

So, I think the compromise is to introduce a `UserWorkspaceBuildParameters` resource via the user object (similar to `UserData`), then grant read permission to the user for their own parameters and the site admin. That way when we look at the permissions in the roles, we can see which can read these parameters, instead of just having to know that Update/UserData is equivalent to Read/BuildParameters.